core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -157,15 +159,24 @@ public void onReady() {}
   public abstract void request(int numMessages);
 
   /**
+   * Equivalent as {@code cancel(null, null)}.
+   */
+  public final void cancel() {
```

Comment:
I can keep `cancel()` abstract and implement `cancel(String, Throwable)` in this abstract class, but it looks wacky and we probably want to fix it. We cannot fix this situation if we want to keep both methods, because there is no deprecation mechanism for overridability. Maybe we should mark `cancel()` as deprecated? I think we are still able to do it before 1.0.